tests/Integration/UI/AdminColumnsParselyStatsTest.php

```diff
@@ -64,6 +65,382 @@ public function tear_down(): void {
+	 * @covers \Parsely\UI\Admin_Columns_Parsely_Stats::enqueue_parsely_stats_script_with_data
+	 * @covers \Parsely\UI\Admin_Columns_Parsely_Stats::get_script_id
+	 *
+	 * @group content-helper
+	 * @group content-helper-admin-stats-column
+	 */
+	public function test_assets_get_enqueued_when_global_filter_is_false_and_feature_filter_is_true(): void {
+		self::assert_enqueued_status( false, true, true );
+	}
+
+	/**
+	 * Verifies that the run() method does not enqueue the assets when the
+	 * global filter is set to true and the feature filter is set to an invalid
+	 * value.
+	 *
+	 * @since 3.9.0
+	 *
+	 * @covers \Parsely\UI\Admin_Columns_Parsely_Stats::get_feature_filter_name
+	 * @covers \Parsely\Content_Helper\Content_Helper_Feature::get_global_filter_name
+	 * @covers \Parsely\UI\Admin_Columns_Parsely_Stats::__construct
+	 * @covers \Parsely\UI\Admin_Columns_Parsely_Stats::run
+	 * @covers \Parsely\UI\Admin_Columns_Parsely_Stats::should_be_enabled
+	 * @covers \Parsely\RemoteAPI\Remote_API_Base::is_user_allowed_to_make_api_call
+	 * @covers \Parsely\Content_Helper\Content_Helper_Feature::is_enabled_by_filters
+	 * @covers \Parsely\UI\Admin_Columns_Parsely_Stats::set_current_screen
+	 * @covers \Parsely\UI\Admin_Columns_Parsely_Stats::is_tracked_as_post_type
+	 * @covers \Parsely\UI\Admin_Columns_Parsely_Stats::enqueue_parsely_stats_script_with_data
+	 * @covers \Parsely\UI\Admin_Columns_Parsely_Stats::get_script_id
+	 *
+	 * @group content-helper
+	 * @group content-helper-admin-stats-column
+	 */
+	public function test_assets_do_not_get_enqueued_when_global_filter_is_true_and_feature_filter_is_invalid(): void {
+		self::assert_enqueued_status( true, array(), false );
+	}
+
+	/**
+	 * Verifies that the run() method enqueues the assets when the global filter
+	 * is set to an invalid value and the feature filter is set to true.
+	 *
+	 * @since 3.9.0
+	 *
+	 * @covers \Parsely\UI\Admin_Columns_Parsely_Stats::get_feature_filter_name
+	 * @covers \Parsely\Content_Helper\Content_Helper_Feature::get_global_filter_name
+	 * @covers \Parsely\UI\Admin_Columns_Parsely_Stats::__construct
+	 * @covers \Parsely\UI\Admin_Columns_Parsely_Stats::run
+	 * @covers \Parsely\UI\Admin_Columns_Parsely_Stats::should_be_enabled
+	 * @covers \Parsely\RemoteAPI\Remote_API_Base::is_user_allowed_to_make_api_call
+	 * @covers \Parsely\Content_Helper\Content_Helper_Feature::is_enabled_by_filters
+	 * @covers \Parsely\UI\Admin_Columns_Parsely_Stats::set_current_screen
+	 * @covers \Parsely\UI\Admin_Columns_Parsely_Stats::is_tracked_as_post_type
+	 * @covers \Parsely\UI\Admin_Columns_Parsely_Stats::enqueue_parsely_stats_script_with_data
+	 * @covers \Parsely\UI\Admin_Columns_Parsely_Stats::get_script_id
+	 *
+	 * @group content-helper
+	 * @group content-helper-admin-stats-column
+	 */
+	public function test_assets_get_enqueued_when_global_filter_is_invalid_and_feature_filter_is_true(): void {
+		self::assert_enqueued_status( 'string', true, true );
+	}
```

Comment:
This is the initial approach I took but I concluded that it does not make things simpler or more concise, as I ended up with more lines of code, tighter coupling and slightly more complexity.

The reason is that we still need to create a function for every test in the child class to add the `covers` DocBlocks there (because they are different for every child class), as well as call the parent testing function from there.

So unless you had a specific methodology in mind to address this, for me the current approach is fine.

Reply:
Currently we have `RemoteAPITest` which we use as extend class in our other tests, all tests inside `RemoteAPITest` class runs for each API seprately so what I am thinking here is that if we have move all of these duplicated tests inside `ContentHelperFeatureTest` then we can achieve same thing with less code and in future also it will save our time.